customer/views.py

```diff
@@ -43,3 +45,21 @@ def post(self, request, *args, **kwargs):
                 return Response(serializer.data, status=status.HTTP_201_CREATED)
         else:
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ListAllCustomersView(generics.ListAPIView):
+    # this is only for testing purposes
+    queryset = CustomUser.objects.all()
+    serializer_class = ListCustomersSerializer
+
+
+class CustomerDetailsView(generics.RetrieveAPIView):
```

Comment:
why are u still using the  "generics.RetrieveAPIView" 
when this is based on the request.user and u anyway overrides the get method 
use generic api view insted for better flow

Reply:
and remove the id lookup